executor/read/stream_modifier.rs

```diff
@@ -200,13 +186,13 @@ impl LimitMapper {
 }
 
 impl StreamModifierResultMapperTrait for LimitMapper {
-    fn map_output(&mut self, subquery_result: Option<FixedBatch>) -> StreamModifierControl {
+    fn map_output(&mut self, subquery_result: Option<FixedBatch>) -> Option<FixedBatch> {
```

Comment:
Why dd you move the other enum to the Option? An Option::Some(empty batch) is stranger than a Control::Retry no?

Reply:
Standardisation. It felt weird for this to return something different.
Some(empty_batch) is less explicit, but it fits fine. It's like a check instruction ruling out all the rows it received.

Reply:
hm ok